src/server/api/user.ts

```diff
@@ -155,17 +136,21 @@ router.post('/url', validateUrls, async (req, res) => {
     }
 
     // Success
-    const result = await transaction((t) =>
-      Url.create(
+    const result = await transaction(async (t) => {
+      const url = Url.create(
         {
           userId: user.id,
           longUrl,
           shortUrl,
           isFile: !!isFile,
         },
         { transaction: t },
-      ),
-    )
+      )
+      if (isFile) {
+        await uploadFileToS3(file.data, shortUrl, file.mimetype)
```

Comment:
is the file extension being saved to the S3 bucket?

Reply:
nope. The S3 bucket only has the shortUrl as the key. the only hint that S3 gets is the ContentType which is obtained from the mimetype argument in this function call. This will let S3 behave accordingly depending on the file type. For example, if it received image/jpeg, it will display the image in the browser tab instead of initiating a download, etc.

Reply:
Interesting. What file types are available?

Reply:
my bad, after digging through docs and web resources, it appears that S3 does not inherently behave differently to different Content-Type headers. Instead this is stored and forwarded to all browsers that attempt to retrieve the object. It is then up to the browsers to behave accordingly, such as displaying the content directly if it's a pdf or image, and initiating a download if it's an executable or a video file, etc.

Quoting from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types), _Important: Browsers use the MIME type, not the file extension, to determine how to process a URL, so it's important that web servers send the correct MIME type in the response's Content-Type header. If this is not correctly configured, browsers are likely to misinterpret the contents of files and sites will not work correctly, and downloaded files may be mishandled._

The list of MIME types are also available in the MDN link above.